src/mlpack/core/optimizers/adam/adam_impl.hpp

```diff
@@ -67,6 +70,9 @@ double Adam<DecomposableFunctionType>::Optimize(arma::mat& iterate)
   // Exponential moving average of gradient values.
   arma::mat m = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
 
+  // Initialize the exponentially weighted infinity norm for AdaMax optimizer.
+  arma::mat u = arma::zeros<arma::mat>(iterate.n_rows, iterate.n_cols);
```

Comment:
Yes, I think may be we can move the initialization of u and v inside the `if ... else` at line 122. What do you suggest?